### DISCUSSION TITLE
Tasmota relay different switching times

### DISCUSSION THREAD

**Discussion by GeorgeWatch** [arendst/Tasmota#17449] [Q&A] [answered]
Hi
I am trying to separate the 1-9 I2C PCF8574 relay on Tasmota. I need relay 1 to close after 1.5s after power is turned on or when manually turned off. This is what I managed to do:
PowerOnState1 5 and PulseTime1 15. (Relay 1 start Tasmota turns on after 1.5s or manual shutdown turns on after 1.5s)
I would need relays 2-9 after switching on after 1s off. Can it be set?
I can't get it to work like that
Thank you.
G.W.

**Comment by sfromis** [comment, marked as answer]
Not clear what you are doing, or what your use case is.

`PowerOnState1 5` should enable a brief power off, returning to on afterwards, using your PulseTime1. Not sure if this is what you wrote....

About relay 2-9, you wrote "after switching on after 1s off", which is not crystal clear. If you mean that you want the same as for relay 1, but with a shorter off-period, that sounds like `PowerOnState2 5` and `PulseTime2 10` - repeat for each relay.

**Reply by GeorgeWatch** [reply]
Thank you for answer. I guess I wrote it in a complicated way.
I am trying to simulate a short button press. About cca 1 second. I don't have relays on the outputs, but optocouplers controlling other elements. I will try to see if it will work according to your description.
G.W.

**Reply by sfromis** [reply]
Physical relay vs optocoupler should make no difference to Tasmota (which is just controlling high/low on a gpio), but details of the circuit you connect to (and how) could potentially mean different behavior.

**Reply by GeorgeWatch** [reply]
I am trying to use ESP-01 Tasmota and I2C to control other devices. The problem is that the PCF8574 generates short pulses when the power is turned on. I tried to eliminate this by using a power transistor that turns on the power for the output elements after a given time until the EPS and PCF are initialized. I ran into a PowerOnState limitation (or I'm reading the documentation wrong). After switching on, all outputs are either On or Off according to the PowerOnState value 0 or 1. It is not possible to define how each output should behave after switching on the power...


translate google :-)
